tests/testTriggers.go

```diff
@@ -4770,6 +4771,105 @@ func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *Event
 	updateMetrics(*event)
 }
 
+// TriggerVolumeUpdate enables to test volume update
+func TriggerVolumeIOProfileUpdate(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer ginkgo.GinkgoRecover()
+	defer endLongevityTest()
+	startLongevityTest(UpdateVolume)
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: UpdateVolume,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+	setMetrics(*event)
+	stepLog := "Validate IO profile update on volumes"
+	context(stepLog, func() {
+		log.InfoD(stepLog)
+		Step(stepLog,
+			func() {
+				//count := 0
+				log.InfoD(stepLog)
+				for count := range [2]int{} {
+					updateIOProfileOnVolumes(contexts, event)
+					for _, ctx := range *contexts {
+						errorChan := make(chan error, errorChannelSize)
+						log.Infof("Validating context: %v", ctx.App.Key)
+						ctx.SkipVolumeValidation = false
+						ValidateContext(ctx, &errorChan)
+						for err := range errorChan {
+							log.Infof("Error: %v", err)
+							UpdateOutcome(event, err)
+						}
+					}
+					if count < 1 {
+						log.InfoD("Wait for 15 minutes before restting back")
+						time.Sleep(15 * time.Minute)
+						log.InfoD("Resetting back")
+					}
+				}
+			})
+
+	})
+	log.InfoD("Update IO profile completed")
+	updateMetrics(*event)
+}
+
+// updateIOProfileOnVolumes this method is responsible for updating IO priority on Volumes.
+func updateIOProfileOnVolumes(contexts *[]*scheduler.Context, event *EventRecord) {
+	for _, ctx := range *contexts {
+		var appVolumes []*volume.Volume
+		var volumeSpec *apios.VolumeSpecUpdate
+		var err error
+		//var requiredIOProfile apios.VolumeSpecPolicy_IoProfile
+		var requiredIOProfile apios.IoProfile
+		appVolumes, err = Inst().S.GetVolumes(ctx)
+		UpdateOutcome(event, err)
+		if len(appVolumes) == 0 {
+			UpdateOutcome(event, fmt.Errorf("found no volumes for app "))
+		}
+		for _, v := range appVolumes {
+			log.InfoD("Getting info from volume: %s", v.ID)
+			appVol, err := Inst().V.InspectVolume(v.ID)
+			if err != nil {
+				log.Errorf("Error inspecting volume: %v", err)
+			}
+			currentIOProfile := appVol.Spec.IoProfile
+			derivedIOProfile := appVol.DerivedIoProfile
+			log.InfoD("Volume: %s Current IO profile : %s, current derived IO profile %s", v.ID, currentIOProfile, derivedIOProfile)
+			if currentIOProfile == apios.IoProfile_IO_PROFILE_AUTO {
```

Comment:
is profile journal not required?

Reply:
We will use AutoJournal, that should be enough. If required we will add it